tests/Functional/HStringMapTest.php

```diff
@@ -39,4 +40,24 @@ public function testStringMapRot13()
         $expected = "sbbone";
         $this->assertEquals($expected, $newString);
     }
+
+    public function testStringWithVariadicArray()
+    {
+        $myArray = new HArray([
+            "apple",
+            "butter",
+            "cookie",
+            "donut",
+            "eclaire",
+            "framboise",
+        ]);
+
+        $oddPairing = function ($sentenceWord, $arrayWord) {
+            return sprintf("-%s:%s-", $sentenceWord, $arrayWord);
+        };
+
+        $pairs = $this->aString->map($oddPairing, $myArray);
+        $expected = "-f:apple--o:butter--o:cookie--b:donut--a:eclaire--r:framboise-";
```

Comment:
Good call. I was certain that this behaved the same way, but without tests, I cannot prove it. :)